src/UI/WB.UI.Designer/Controllers/Api/WebTester/WebTesterController.cs

```diff
@@ -86,14 +89,24 @@ public IActionResult QuestionnaireAsync(string token)
             }
         }
 
+        private Guid GetOriginalQuestionnaireId([DisallowNull] Guid? questionnaireId)
+        {
+            var anonymousQuestionnaire = designerDbContext.AnonymousQuestionnaires.FirstOrDefault(a => a.IsActive == true

```

Comment:
I think it would be a bit more efficient if we do select only by Primary key and check IsActive in code.